insights/client/__init__.py

```diff
@@ -255,22 +255,54 @@ def install(self, new_egg):
         """
         returns (dict): {'success': True if the new core was installed successfull else False}
         """
+        import os
+        from shutil import copyfile
+
         success = False
         if not new_egg:
-            logger.debug('Must provide a valid Core installation path.')
-            return {'success': success}
+            the_message = 'Must provide a valid Core installation path.'
+            logger.debug(the_message)
+            return {'success': success, 'message': the_message}
 
         logger.debug("Installing the new Core %s", new_egg)
 
-        from setuptools.command import easy_install
+        # Make sure /var/lib/insights exists
         try:
-            easy_install.main(["-U", new_egg])  # WHY DOESNT THIS RETURN ANYTHING
-            success = True
-        except:
-            logger.debug("There was an issue installing the new Core.")
-            logger.debug("Please ensure a proper path to the Core.")
-            logger.debug("Please ensure you are running as root, or a user with root privileges.")
-
+            if not os.path.isdir(constants.insights_core_lib_dir):
+                logger.debug("Creating directory %s for the Core." % (constants.insights_core_lib_dir))
+                os.mkdir(constants.insights_core_lib_dir)
+        except OSError:
+            message = "There was an error creating %s for Core installation." % (constants.insights_core_lib_dir)
+            raise OSError(message)
+
+        # Copy the NEWEST egg to /var/lib/insights/last_stable.egg
+        old_newest_egg = constants.insights_core_rpm
```

Comment:
I had installed the 'rpm.egg' as 'last_stable.egg' as an additional fallback mechanism but that logic can be removed. I believe the logic should be (as you stated), if there is a 'newest.egg' then copy/move that to 'last_stable.egg,' then copy/move the egg that was fetched and verified in /tmp to 'newest.egg.' I'm not sure exactly how we track if the 'newest.egg's last run was actually successfully. First we would need to define what successful was, I assume that would be that it made it through all essential steps (collecting and uploading). Then maybe after a successful upload we keep track of a dot file in /etc/insights-client. Then, however, we would have to add some logic to link that dot file to that actual egg file.